parquet-variant-compute/src/variant_array.rs

```diff
@@ -439,6 +440,20 @@ impl From<VariantArray> for ArrayRef {
     }
 }
 
+impl<'m, 'v> FromIterator<Option<Variant<'m, 'v>>> for VariantArray {
```

Comment:
If we do this:
```suggestion
impl<'m, 'v, V: Into<Variant<'m, 'v>> FromIterator<Option<V>> for VariantArray {
```
then we can create a variant array from anything that converts cleanly to variant (e.g. i64 values or &str)?

Reply:
This is mostly going to be used by test code, and a homogenous set of e.g. i64 is a pretty common use case. But I don't have strong feelings either way.